scripts/superdesk/editor/spellcheck/spellcheck.js

```diff
@@ -77,6 +83,100 @@ function SpellcheckService($q, api, dictionaries) {
         angular.extend(dict.content, item.content || {});
     }
 
+    function getNonSentenceWords(textContent, currentOffset) {
+        // words contains one or more periods in content. e.g., E.N.T., Aborig. or etc.
+        var reAbbreviations = /\w+(?:\.\w*)+/g;
+        var abbreviationWords = _.words(textContent, reAbbreviations);
+
+        // consider only abbreviations in content that found in dictionary
+        var _abbrevArr = [];
+        _.forEach(abbreviationWords, function(abbrevWord) {
+            _.filter(abbreviationList, function(item) {
+                if (item === abbrevWord) {
+                    _abbrevArr.push(abbrevWord);
+                }
+            });
+        });
+        var _abbreviationString = _abbrevArr.join('|');
+
+        // Prepare non sentence words
+        var match, wordIndex;
+        var nonSentenceWords = [];
+
+        // Words that follows abbreviations in content
+        var _reNonSentenceWords = '\\s+(?:' + _abbreviationString + ')(\\s+\\w+)';
+        var reNonSentenceWords = new RegExp(_reNonSentenceWords, 'g');
+        while ((match = reNonSentenceWords.exec(textContent)) != null) {
+            wordIndex = match.index + match[0].indexOf(_.trim(match[1]));
+            nonSentenceWords.push({
+                word: _.trim(match[1]),
+                index: currentOffset + wordIndex
+            });
+        }
+
+        return nonSentenceWords;
+    }
+
+    function testExpression(textContent, currentOffset) {
```

Comment:
pls add some comment to those functions, not exactly clear what it does by its name